.github/workflows/docs.yml

```diff
@@ -1,56 +1,57 @@
-#
-# Copyright (c) 2022 alandefreitas (alandefreitas@gmail.com)
-#
-# Distributed under the Boost Software License, Version 1.0.
-# https://www.boost.org/LICENSE_1_0.txt
-#
+name: Publish Docs
 
-name: Docs Generator
-
-on: push
+on:
+  push:
+    branches:
+      - master
 
 jobs:
   generateDOC:
-    name: Docs Generator
-    runs-on: ubuntu-20.04
+    name: Publish Docs
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Generate Table of Contents
-        uses: technote-space/toc-generator@v2
-        with:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          MAX_HEADER_LEVEL: 3
-          FOLDING: true
       - name: Update Contributors
-        if: github.ref == 'refs/heads/master'
-        uses: akhilmhdh/contributors-readme-action@v2.0.1
+        uses: akhilmhdh/contributors-readme-action@v2.3
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           image_size: 100
           columns_per_row: 6
-      - name: Download mdsplit
+          readme_path: docs/acknowledgments.md
+        continue-on-error: true
+      - name: Install doxygen
+        run: sudo apt-get install doxygen
+      - name: Generate doxygen XML
+        run: doxygen
+        working-directory: ./docs
+      - name: Download doxybook2
         uses: carlosperate/download-file-action@v1.0.3
-        id: download-mdsplit
+        id: download-doxybook2
         with:
-          file-url: 'https://github.com/alandefreitas/mdsplit/releases/download/v0.1.1/Executable.Linux.zip'
-          file-name: 'mdsplit.zip'
-          location: '.'
-      - name: Unzip mdsplit
+          file-url: 'https://github.com/matusnovak/doxybook2/releases/download/v1.4.0/doxybook2-linux-amd64-v1.4.0.zip'
+          file-name: 'doxybook2.zip'
+          location: './docs'
+      - name: Unzip doxybook2
         run: |
-          unzip mdsplit.zip
-          rm -f mdsplit.zip
-          sudo chmod +x mdsplit
+          unzip doxybook2.zip
+          rm -f doxybook2.zip
+          sudo chmod +x ./bin/doxybook2
+          mkdir reference
           ls
-      - name: Generate Documentation Source
```

Comment:
The documentation currently includes one file for each section from the README.md. Sinces we are getting rid of mdspli, we need to split the `README.md` (your `index.md`) locally and push these pages to the documentation. 

Then the new ./README.md can become just a link to the documentation, as in `futures`.

Reply:
I'm not quite sure I follow you. So you're saying that the new README can be as in futures (just with a link there), and we need to add new .md files for each new documentation section that we might have. Is that right?

Reply:
Now I see it here, I'll split the README.md :)